### PR TITLE
app-project: Create Storybook global decorator for Grommet theme

### PR DESCRIPTION
**Pull request by goplayoutside3**
This includes #4417, please review that one first.

## Package
app-project

## Linked Issue and/or Talk Post
Towards upgrading all of FEM to Storybook version 7

## Describe your changes
Similar to the Storybook changes in #4409, this PR creates a [global decorator](https://storybook.js.org/docs/react/essentials/toolbars-and-globals) for our Grommet theme. When viewing any component story, you can toggle the theme between light and dark via the top toolbar.

Note that by removing the Grommet theme provider component from individual stories, it causes an individual app-project component to render in a more "raw form". For instance, in the Stat story the Stat component no longer has any extra padding, margin, or extra background styling. Its story renders the component as it exists irl. Another example is how TeamMember has a transparent background, which is intentional because it always exists inside an AboutPage component. 

I want to include the above note because minor styling differences between master and this branch did not go unnoticed. If the app-project component looked very different once its Grommet theme provider was removed from the individual story file, I refactored the story to include an extra `Box` or extra `pad` property.

## How to Review

Code changes are only to the storybook files.

Therefore, the best way to review is to run `yarn storybook` on this branch for app-project. All component stories should load and function almost exactly as the published [Frontend Storybook](https://zooniverse.github.io/front-end-monorepo/). Any minor fixes that require further code changes to the actual component files or unit tests should go in another PR. This PR is scoped to only stories.js files.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Refactoring
- [ ] The PR creator has described the reason for refactoring
- [ ] The refactored component(s) continue to work as expected

### CODE REVIEW

**Comment by shaunanoordin** [comment]
Ah, butts - there are 10 merge conflicts at the moment. I'll continue reviewing since this is similar enough to 4409 that I know exactly what to look out for, but I'll take a second pass after the conflicts get de-conflicted.

File: packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/RecentSubjects/RecentSubjects.stories.js [hunk 1]
```diff
@@ -105,56 +140,35 @@ export default {
   component: RecentSubjects,
   args: {
     dark: false
```
Comment (shaunanoordin) [inline packages/app-project/src/screens/ProjectHomePage/components/ZooniverseTalk/components/RecentSubjects/RecentSubjects.stories.js:142]: I think this `args.dark` needs to be removed. I'm seeing an unusable toggle for 'dark' under Controls, for all Stories under Screens -> Project Home -> Recent Subjects -> *

(Not to be confused with Screens -> Classify -> Recent Subjects -> * , which look fine)

**Review by shaunanoordin — APPROVED** [review]
## PR Review

Package: `app-project`
Similar to: #4409

This PR updates the Storybook Stories for every component in lib-react-components, rewriting the components into a more standardised format and moving some general settings (notably, light/dark theme) into a more appropriate **global/default config.**

Changes noted:
- Global config: `.storybook/preview.js` has been updated.
  - Now, every component's story _automagically_ comes wrapped in a `<Grommet>` super-parent with the theme prop, etc.
  - In Storybook, **light theme/dark theme** can now be toggled via the new "Grommet Theme" dropdown.
- ❕ Surprisingly, unlike 4409, I'm not seeing Stories where the READMES have to be removed. I assume none of the Components in app-project use composeStory() to include the Stories into their tests.

Code read:
- Changes here are pretty straightforward, even though a large number of files were changed.
- Only the update to `.storybook/preview.js` is noteworthy; all other Components simply had now-redundant code (and/or redundant sub-stories) stripped out.

### Testing

Tested on localhost with macOS + Chrome.

- Ran `yarn storybook` and examined a random sample of components in http://localhost:6007/
- Tried switching between Light mode and Dark mode via the Grommet Theme dropdown. Component visuals updated accordingly.

Everything looks fine, though on a minor note, Project Home -> Recent Subjects -> * Stories have a loose 'dark' arg still running around.

### Status

LGTM 👍 I'm Approving this PR since it's pretty clear what it's doing, though this PR still needs have some conflicts resolved before it's ready to merge.

**Comment by goplayoutside3** [comment]
@shaunanoordin thanks for the review! Conflicts should be resolved now if you'd like a second look 👍

**Pull request merged**